src/fe-fuzz/irc/core/event-get-params.c

```diff
@@ -0,0 +1,86 @@
+/*
+ event-get-params.c : irssi
+
+    Copyright (C) 2017 Joseph Bisch
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "module.h"
+#include "modules-load.h"
+#include "levels.h"
+#include "../fe-text/module-formats.h" // need to explicitly grab from fe-text
+#include "themes.h"
+#include "core.h"
+#include "fe-common-core.h"
+#include "args.h"
+#include "printtext.h"
+#include "irc.h"
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+int LLVMFuzzerInitialize(int *argc, char ***argv) {
+	core_register_options();
+	fe_common_core_register_options();
+	/* no args */
+	args_execute(0, NULL);
+	core_preinit((*argv)[0]);
+	core_init();
+	fe_common_core_init();
+	theme_register(gui_text_formats);
+	module_register("core", "fe-fuzz");
+	return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+	if (size < 1) {
+		return 0;
+	}
+	uint8_t count = *data;
+	char *copy = (char *)malloc(sizeof(char)*(size-1+1));
```

Comment:
Holy redundancy. Why the size-1+1?

sizeof(char) is 1 unless you're dealing with people who aren't fun at parties.

I wanted to say "No need to cast mallocs" but I'm not actually sure this is C and not C++. Is this C++?

Also, just use g_strndup?

Reply:
The size-1+1 is because the first byte of data is already used, so we aren't going to use it again for the string. Then plus one for the terminator. Yeah, I could just do size, but I was trying to make the calculation clear.

I don't know why I'm casting the malloc. I copied that from the original fuzzer in fe-fuzz. Maybe at one point fe-fuzz just straight up used CXX for everything, but now it uses CC for compiling and then CXX for linking. So it does work without the cast.